### PR TITLE
Add ndim attribute for pint-pandas compatibility

### LINKED ISSUE #1599: Pint-Pandas support
Since pandas 1.4 https://github.com/pandas-dev/pandas/pull/44626, the ndim attribute of a object is used to identify the object as a scalar. (thanks @burnpanck!) 

Adding an ndim attribute that defaults to 0 for scalars with magitudes that don't have an ndim attribute defined, along with adding DataFrame to pint's upcast types allows pint-pandas to pass all pandas' extension tests! https://github.com/hgrecco/pint-pandas/pull/133

I've added these changes here https://github.com/hgrecco/pint/pull/1596. It would be great to also have a new pint version for pint-pandas to require once these changes are merged.

### PR DESCRIPTION
**Pull request by andrewgsavage**
- [x] Closes #1599 
- [x] Executed ``pre-commit run --all-files`` with no errors
- [x] The change is fully covered by automated unit tests
- [x] Documented in docs/ as appropriate
- [x] Added an entry to the CHANGES file

### CODE REVIEW

File: CHANGES [hunk 1]
```diff
@@ -11,6 +11,7 @@ Pint Changelog
   (Issue #1030, #574)
 - Added angular frequency documentation page.
 - Move ASV benchmarks to dedicated folder. (Issue #1542)
+- An ndim attribute has been added to Quantity for pint-pandas compatability.
```
Comment (edouardberthe) [inline CHANGES:14]:
```suggestion
- An ndim attribute has been added to Quantity for pint-pandas compatibility. 
```

Comment (andrewgsavage) [reply CHANGES]: good spot

**Comment by andrewgsavage** [comment]
This is ready for review.

**Comment by hgrecco** [comment]
awesome, thanks!

**Pull request merged**